core/src/main/java/de/bwaldvogel/mongo/oplog/OplogDocument.java

```diff
@@ -0,0 +1,130 @@
+package de.bwaldvogel.mongo.oplog;
+
+import java.time.Instant;
+import java.util.UUID;
+
+import de.bwaldvogel.mongo.bson.Bson;
+import de.bwaldvogel.mongo.bson.BsonTimestamp;
+import de.bwaldvogel.mongo.bson.Document;
+
+public class OplogDocument implements Bson {
+    private Document document;
+
+    public OplogDocument() {
+        document = new Document();
+        withProtocolVersion(2L);
+    }
+
+    public Document toDocument() {
+        return document;
+    }
+
+    public BsonTimestamp getTimestamp() {
+        return (BsonTimestamp) document.get("ts");
+    }
+
+    public OplogDocument withTimestamp(BsonTimestamp timestamp) {
+        document.put("ts", timestamp);
+        return this;
+    }
+
+    public long getT() {
+        return (Long) document.get("t");
+    }
+
+    public OplogDocument withT(long t) {
+        document.put("t", t);
+        return this;
+    }
+
+    public long getHash() {
+        return (Long) document.get("h");
+    }
+
+    public OplogDocument withHash(long h) {
+        document.put("h", h);
+        return this;
+    }
+
+    public long getProtocolVersion() {
+        return (Long) document.get("v");
+    }
+
+    public OplogDocument withProtocolVersion(long protocolVersion) {
+        document.put("v", protocolVersion);
+        return this;
+    }
+
+    public OperationType getOperationType() {
+        return OperationType.valueOf((String) document.get("op"));
```

Comment:
Probably you want a mapping between OperationType name (`DELETE`) and value on the wire (`d`)

Reply:
See https://github.com/bwaldvogel/mongo-java-server/pull/131#discussion_r410866726